src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/custom.py

```diff
@@ -0,0 +1,332 @@
+                       slot=None):
+    from azure.mgmt.web.models import (FileSystemApplicationLogsConfig, ApplicationLogsConfig,
+                                       SiteLogsConfig, HttpLogsConfig,
+                                       FileSystemHttpLogsConfig, EnabledConfig)
+    client = web_client_factory()
+    #TODO: ensure we call get_site only once
+    site = client.sites.get_site(resource_group, name)
+    location = site.location
+
+    application_logs = None
+    if application_logging:
+        if application_logging == 'off': #TODO share the same choice list in _params.py
+            level = 'Off'
+        elif level is None:
+            level = 'Error'
+        fs_log = FileSystemApplicationLogsConfig(level)
+        application_logs = ApplicationLogsConfig(fs_log)
+
+    http_logs = None
+    if web_server_logging:
+        enabled = web_server_logging != 'off'
+        #100 mb max log size, retenting last 3 days. Yes we hard code it, portal does too
+        fs_server_log = FileSystemHttpLogsConfig(100, 3, enabled)
+        http_logs = HttpLogsConfig(fs_server_log)
+
+    detailed_error_messages_logs = (None if detailed_error_messages is None
+                                    else EnabledConfig(detailed_error_messages == 'on'))
+    failed_request_tracing_logs = (None if failed_request_tracing is None
+                                   else EnabledConfig(failed_request_tracing == 'on'))
+    site_log_config = SiteLogsConfig(location,
+                                     application_logs=application_logs,
+                                     http_logs=http_logs,
+                                     failed_requests_tracing=failed_request_tracing_logs,
+                                     detailed_error_messages=detailed_error_messages_logs)
+    if slot is None:
+        return client.sites.update_site_logs_config(resource_group, name, site_log_config)
+    else:
+        return client.sites.update_site_logs_config_slot(resource_group, name,
+                                                         site_log_config, slot)
+
+def config_slot_auto_swap(resource_group, webapp, slot, auto_swap_slot=None, disable=None):
+    client = web_client_factory()
+    site_config = client.sites.get_site_config_slot(resource_group, webapp, slot)
+    site_config.auto_swap_slot_name = '' if disable else (auto_swap_slot or 'production')
+    return client.sites.update_site_config_slot(resource_group, webapp, site_config, slot)
+
+def get_streaming_log(resource_group, name, provider=None, slot=None):
+    client = web_client_factory()
+    repo_url = _get_repo_url(client, resource_group, name, slot)
+    streaming_url = repo_url + '/logstream'
+    import time
+    if provider:
+        streaming_url += ('/' + provider.lstrip('/'))
+
+    user, password = _get_site_credential(client, resource_group, name)
+    t = CurlThread(streaming_url, user, password)
+    t.daemon = True
```

Comment:
>t.daemon = True [](start = 4, length = 15)

From py docs: Note:
 Daemon threads are abruptly stopped at shutdown. Their resources (such as open files, database transactions, etc.) may not be released properly. If you want your threads to stop gracefully, make them non-daemonic and use a suitable signalling mechanism such as an Event.
 
Is this a concern here?